pkg/test/case.go

```diff
@@ -397,9 +402,17 @@ func (t *Case) determineNamespace() *namespace {
 	if t.PreferredNamespace == "" {
 		ns.Name = fmt.Sprintf("kuttl-test-%s", petname.Generate(2, "-"))
 		ns.AutoCreated = true
+	} else {
+		exist, err := t.NamespaceExists(t.PreferredNamespace)
+		if err != nil {
+			return nil, fmt.Errorf("found namespace %s err, %s", t.PreferredNamespace, err.Error())
+		}
+		if !exist {
+			ns.AutoCreated = true
+		}
 	}
 	// if we have a preferred namespace, we do NOT auto-create
```

Comment:
This comment is now stale, and in the wrong place...
Please move it into the else block, and change to say something like:
```suggestion
	// if we have a preferred namespace, and it already exists, we do NOT auto-create
```